great_expectations/render/renderer/v3/suite_profile_notebook_renderer.py

```diff
@@ -33,64 +34,52 @@ def __init__(
             batch_request=batch_request
         )
 
-        self.batch_request = batch_request
-
-        self.validator = context.get_validator(
+        self._batch_request = batch_request
+        self._validator = context.get_validator(
             batch_request=BatchRequest(**batch_request),
             expectation_suite_name=expectation_suite_name,
         )
 
-        self.expectation_suite_name = self.validator.expectation_suite_name
+        self._profiler_name = profiler_name
+        self._expectation_suite_name = self._validator.expectation_suite_name
 
     # noinspection PyMethodOverriding
-    def add_header(self) -> None:
-        self.add_markdown_cell(
-            markdown=f"""# Initialize a new Expectation Suite by profiling a batch of your data.
-This process helps you avoid writing lots of boilerplate when authoring suites by allowing you to select columns and other factors that you care about and letting a profiler write some candidate expectations for you to adjust.
+    def render(self) -> nbformat.NotebookNode:
+        self._notebook = nbformat.v4.new_notebook()
```

Comment:
Much of this file is just reorganization to put the key methods at the top (`render` and `render_to_disk`).

The core logic you want to review is in this `render` method and the various calls it makes.